x/mint/keeper/keeper.go

```diff
@@ -101,3 +103,83 @@ func (k Keeper) MintCoins(ctx context.Context, newCoins sdk.Coins) error {
 func (k Keeper) AddCollectedFees(ctx context.Context, fees sdk.Coins) error {
 	return k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleName, k.feeCollectorName, fees)
 }
+
+func (k Keeper) DefaultMintFn(ctx context.Context, env appmodule.Environment, minter *types.Minter, epochId string, epochNumber int64) error {
```

Comment:
personal preference: it is ok to have this defined at the keeper but an independent method like `ProvideExampleMintFn` would be nicer to compare and test.

Reply:
My idea here was to leave this as untouched as possible. I'd prefer if we merge this as is and then we can do more improvements on it later. We could even reduce the amount of dependencies needed